src/tools.ts

```diff
@@ -268,3 +268,31 @@ export function changeFirstLanguage(
   exploreNode(expr);
   return expr;
 }
+
+/**
+ * Tst if a string matches the pattern "{name:xx}" in a exact way or is a loose way (such as "foo {name:xx}")
+ */
+export function checkNamePattern(str: string): { contains: boolean; exactMatch: boolean } {
+  const regex = /\{name:[a-zA-Z]{2}\}/;
```

Comment:
There are languages with code not matching this regex: `name:zh-Hans`, `name:latin`, ... (https://github.com/maptiler/maptiler-client-js/blob/main/src/language.ts)

Reply:
Oh true! I'll fix that!

Reply:
Addressed in the last commit